### PR TITLE
[java] Support groovy installations

### PR DESCRIPTION
**Pull request by Ash258**
Tests are failing for some reason with version 19. I used 21 instead to make tests work again

image

### CODE REVIEW

**Comment by Ash258** [comment]
@microsoft-github-policy-service agree

File: src/java/devcontainer-feature.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "id": "java",
-  "version": "1.2.2",
+  "version": "1.2.3",
```
Comment (samruddhikhandale) [inline src/java/devcontainer-feature.json:3]: As we are adding a new Feature option, can we bump the `minor` instead?

File: src/java/devcontainer-feature.json [hunk 2]
```diff
@@ -74,6 +75,22 @@
       ],
       "default": "latest",
       "description": "Select or enter an Ant version"
+    },
+    "installGroovy": {
+      "type": "boolean",
+      "default": false,
+      "description": "Install Groovy, powerful, optionally typed and dynamic language"
```
Comment (samruddhikhandale) [inline src/java/devcontainer-feature.json:82]:
```suggestion
      "description": "Install Groovy, powerful, optionally typed and dynamic language with static-typing and static compilation capabilities"
```

File: test/java/scenarios.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
         "image": "ubuntu:focal",
         "features": {
             "java": {
-                "version": "19",
```
Comment (samruddhikhandale) [inline test/java/scenarios.json:6]: This fails on `main` branch as well, hence, the PR changes are unrelated. See https://github.com/devcontainers/features/actions/runs/7562535809/job/20593192063#step:4:1092

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Looks great, thank you for adding the super helpful Feature option. 

Left minor comments, besides that looks great!

File: test/java/scenarios.json (hunk 3 above)
Comment (Ash258) [reply test/java/scenarios.json:6]: Should I remove these changes?

Comment (samruddhikhandale) [reply test/java/scenarios.json:6]: Should be fine, thank you!

**Pull request merged**